lib/pubid/iso/transformer.rb

```diff
@@ -44,17 +44,21 @@ class Transformer < Parslet::Transform
 
     rule(type: simple(:type)) do
       russian_type = Renderer::Russian::TYPE.key(type.to_s)
-      { type: russian_type&.to_s ||
+      { type: russian_type&.downcase&.to_sym ||
         case type
         # XXX: can't put 2 alternative Russian translations to dictionary, temporary hack
-        when "GUIDE", "Руководства"
-          "Guide"
-        when "ТС"
-          "TS"
-        when "ТО"
-          "TR"
+        when "GUIDE", "Guide", "Руководства"
+          :guide
+        when "ТС", "TS"
+          :ts
+        when "ТО", "TR"
+          :tr
         when "Directives Part", "Directives, Part", "Directives,"
-          "DIR"
+          :dir
+        when "PAS"
+          :pas
+        when "DPAS"
```

Comment:
DPAS is "Draft of PAS", it is not a type by itself, it belongs to "PAS".

Reply:
> DPAS is "Draft of PAS", it is not a type by itself, it belongs to "PAS".

How should I store draft internally? Do we have any other cases where we need to use "draft"?

Reply:
I don't understand the question. PAS is just like TR or TS. DPAS is just like DTR or DTS. Just treat them the same way?

Reply:
"DIS" itself is confusing, because as you see here: https://user-images.githubusercontent.com/11865/194478303-69dd1861-27d3-4307-8b61-02bb08fb2a1d.jpeg

Because "DIS" means "Draft International Standard". DTS means "Draft Technical Specification", etc. The "D" prefix means different stages for different types. Technically, there is no "DIS" stage for TR, TS and PAS.

Reply:
@ronaldtse it's still not clear to me what to do with DTS, DTR, DPAS. How are we going to treat "D" prefix?
So it's draft, ok. But it affects stage, right?
Does DTR and TR have different stage?
From here: https://user-images.githubusercontent.com/11865/194478303-69dd1861-27d3-4307-8b61-02bb08fb2a1d.jpeg
I can see that "DTR" can have stages from 30 to 50.
In current implementation we use stage 40 for DTS.
Maybe we should just mark it as "draft" without applying any stage? Because we don't know exact stage, but we can raise an error if supplied stage different from range 30 to 50.

Reply:
"DTR" is the type-stage-abbreviation of "TR" in stage 40 to 50. Notice that stage code 30 is a "CD TR".

Perhaps this is useful as a clarification of the type-stage definitions. i.e.

```ruby
TECHNICAL_REPORT = {
  type_stage: {
    pwitr: {
      abbr: "PWI TR",
      name: "Preliminary Work Item Technical Report",
      harmonized_stages: [ "10.00", "10.20", ... ]
    },
    awitr: {
      abbr: "AWI TR",
      name: "Approved Work Item Technical Report",
      harmonized_stages: [ "10.99" ]
    },
    wdtr: {
      abbr: "WD TR",
      name: "Working Draft Technical Report",
      harmonized_stages: [ 20 ]
    },
    wdtr: {
      abbr: "WD TR",
      name: "Working Draft Technical Report",
      harmonized_stages: [ 20 ]
    },
    cdtr: {
      abbr: "CD TR",
      name: "Committee Draft Technical Report",
      harmonized_stages: [ 30 ]
    },
    dtr: {
      abbr: "DTR",
      name: "Draft Technical Report",
      harmonized_stages: [ 40, 50 ]
    },
    tr: {
      abbr: "TR",
      name: "Technical Report",
      harmonized_stages: [ "60" ] # etc.
  }
}
```

We can define this for every PubID document type.

Reply:
> "DTR" is the type-stage-abbreviation of "TR" in stage 40 to 50. Notice that stage code 30 is a "CD TR".
> 
> Perhaps this is useful as a clarification of the type-stage definitions. i.e.
> 
> ```ruby
> TECHNICAL_REPORT = {
>   type_stage: {

@ronaldtse I'm trying to figure out how I'm going to store "type stage".
If I parsed "ISO DTR 1234" I should store whole "DTR" without splitting it to type and stage, right?
In that case we have to get rid of type and stage and introduce "type stage" instead... But...

```ruby
> type_stage = TypeStage.new(abbr: :DTR)
> type_stage.type.to_s
=> "TR"
> type_stage.stage
=> nil
> type_stage.type_stage
=> "DTR"
> type_stage.to_s
=> "DTR"
> type_stage = TypeStage.new(type: :tr, stage: :cd)
> type_stage.stage.to_s
=> "CD"
> type_stage.type_stage
=> nil
> type_stage.to_s
=> "CD TR"
> type_stage.name
=> "Committee Draft Technical Report"
> pubid = Identifier.parse("ISO DTR 1234")
# (pubid.type is a proxy to pubid.type_stage.type)
> pubid.type == pubid.type_stage.type
=> true
# (pubid.stage is a proxy to pubid.type_stage.stage)
> pubid.stage == pubid.type_stage.stage
=> true
> pubid.type_stage.to_s
=> "DTR"
```

Does it make sense?

Reply:
> If I parsed "ISO DTR 1234" I should store whole "DTR" without splitting it to type and stage, right?

Don't think so -- why is that necessary? 

There are document types, document stages, and the type-stages. 
* For a certain type, a certain stage, there is 1 corresponding type-stage.
* For every type-stage, there is 1 corresponding type, 1 (or more) corresponding stages (i.e. 60.20 and 60.60 should have the same type-stage.

Reply:
> > If I parsed "ISO DTR 1234" I should store whole "DTR" without splitting it to type and stage, right?
> 
> Don't think so -- why is that necessary?
> 
> There are document types, document stages, and the type-stages.
> 
> * For a certain type, a certain stage, there is 1 corresponding type-stage.
> * For every type-stage, there is 1 corresponding type, 1 (or more) corresponding stages (i.e. 60.20 and 60.60 should have the same type-stage.

@ronaldtse then from what I should build "DTR"? How should I store "DTR" for identifier?
As I understand, your point is not to store "DTR" as "DIS" and "TR" because "DIS" actually doesn't correspond to "DTR".
So, do I build "DTR" from "TR" and what else it should be?

Reply:
> So, do I build "DTR" from "TR" and what else it should be?

I already gave you a sample structure above. Let's call this "TypedStage". There is "document type", "harmonized stage", and "typed stage". A "typed stage" links to the "document type" and "harmonized stage".

You will store "DTR" as a typed stage. The typed stage will link to "document type TR" and "harmonized stage 40.xx".